### PR TITLE
Add script to build a single example

### PR DESCRIPTION
**Pull request by anghelcovici**
This is a small script that allows to build only one of the examples,
either by entering it as an arg or selecting it from a list.
Additionally, it allows us to build the C++ code in debug mode, if
needed.

### CODE REVIEW

File: scripts/build_example [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+readonly SCRIPTS_DIR="$(dirname "$0")"
+
+COMPILATION_MODE="opt"
+
+while getopts "hd" opt; do
+    case "$opt" in
+    h)
+        echo -e "Usage: ./scripts/build_example [-d] [example_name]
+  -d    Build using debug mode
+  -h    Pritn this message"
```
Comment (tiziano88) [inline scripts/build_example]:
```suggestion
  -h    Print this message"
```

File: scripts/build_example [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+done
+
+readonly NAME="${@:$OPTIND:1}"
+
+if [[ -z "${NAME}" ]]; then
+    readonly EXAMPLES="$(find examples -mindepth 2 -maxdepth 2 -type f -name Cargo.toml | cut -d'/' -f2)"
+    PS3='Choose example to build: '
+
+    options=($EXAMPLES)
+    select opt in "${options[@]}";
```
Comment (tiziano88) [inline scripts/build_example]: What is this sorcery?

Comment (tiziano88) [reply scripts/build_example]: (no need to reply)

File: scripts/build_example [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+set -o errexit
+
+readonly SCRIPTS_DIR="$(dirname "$0")"
+
+COMPILATION_MODE="opt"
+
+while getopts "hd" opt; do
+    case "$opt" in
+    h)
+        echo -e "Usage: ./scripts/build_example [-d] [example_name]
```
Comment (tiziano88) [inline scripts/build_example]:
```suggestion
        echo -e "Usage: $0 [-d] [example_name]
```

Comment (tiziano88) [reply scripts/build_example]: Not sure this works correctly, please try it first :)

File: scripts/build_example [hunk 4]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+readonly SCRIPTS_DIR="$(dirname "$0")"
+
+COMPILATION_MODE="opt"
```
Comment (tiziano88) [inline scripts/build_example]:
```suggestion
COMPILATION_MODE='opt'
```

File: scripts/build_example [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+while getopts "hd" opt; do
+    case "$opt" in
+    h)
+        echo -e "Usage: ./scripts/build_example [-d] [example_name]
+  -d    Build using debug mode
+  -h    Pritn this message"
+        exit 0
+        ;;
+    d)  
+        COMPILATION_MODE="dbg"
```
Comment (tiziano88) [inline scripts/build_example]:
```suggestion
        COMPILATION_MODE='dbg'
```

File: scripts/build_example (hunk 3 above)
Comment (anghelcovici) [reply scripts/build_example]: Works!

File: scripts/build_example (hunk 1 above)
Comment (anghelcovici) [reply scripts/build_example]: words are hard ..

File: scripts/build_example (hunk 2 above)
Comment (anghelcovici) [reply scripts/build_example]: Magic :)

**Comment by anghelcovici** [comment]
@tiziano88 - apparently the cla bot complains about your account :)

**Pull request merged**